runner.py

```diff
@@ -637,18 +638,39 @@ def setup_networks(self):
             self.__networks.append(network)
             self.__join_default_network = True
 
+    # Order service names based on 'depends_on'
+    def order_service_names(self, order_array, service_name):
```

Comment:
since you are using a recursin approach this has to be guarded for a cycle.

Example:
```docker
---
services:
  test-container-1:
    image: alpine
    depends_on:
      - test-container-2
      - test-container-3
  test-container-2:
    image: alpine
  test-container-3:
    image: alpine
    depends_on:
      - test-container-4
  test-container-4:
    image: alpine
    depends_on:
      - test-container-2
      - test-container-4
```

If you feed this file into `docker-compose` it will fail with: 
```log
cycle found: test-container-4 -> test-container-4
```

The python code will fail with:
```log
Traceback (most recent call last):
  File "/private/tmp/test.py", line 27, in <module>
    order_of_service_names = order_service_names(order_of_service_names, service_name)
  File "/private/tmp/test.py", line 14, in order_service_names
    order_array = order_service_names(order_array, dep)
  File "/private/tmp/test.py", line 14, in order_service_names
    order_array = order_service_names(order_array, dep)
  File "/private/tmp/test.py", line 14, in order_service_names
    order_array = order_service_names(order_array, dep)
  [Previous line repeated 995 more times]
  File "/private/tmp/test.py", line 13, in order_service_names
    if dep not in order_array:
RecursionError: maximum recursion depth exceeded in comparison
```

This needs a cycle protection